### ISSUE TITLE
Message filters don't seem to export efficiently

### ISSUE THREAD

**Issue by Hourglass-Hayden** [Tyrrrz/DiscordChatExporter#1048]
### Version

2.40.0

### Flavor

GUI (Graphical User Interface)

### Export format

HTML

### Details

I wanted to do an export of messages from a specific account in a server, so I put in a message filter "from:[user]" (no quotes).
Although it did start an export of what I wanted, it seemed like it was scanning every single message in the server (this server has millions of messages) instead of doing a kind of search within discord & then grabbing those messages. 
Due to how big that server was I had to cancel it because it'd probably take days to finish.

### Steps to reproduce

- Input user token, select channel(s) in a server with a ton of messages
- Type `from:[user]` into Message Filter box (preferably user with few messages)
- It should take a very long time to export just a few msgs from that user

**Comment by Tyrrrz** [comment]
Yes, the filters can only be applied on each message after it's been pulled from the API. They don't affect what gets downloaded, rather what gets saved. That's just how it is.
